### ISSUE TITLE
Problem when Blazor WebAssembly 3.2.0 Preview 4 is not hosted in IIS Express

### ISSUE THREAD

**Issue by Liversage** [dotnet/aspnetcore#20998]
I'm following the directions in the blog post [Blazor WebAssembly 3.2.0 Preview 4 release now available](https://devblogs.microsoft.com/aspnet/blazor-webassembly-3-2-0-preview-4-release-now-available/) but I'm having an issue that has also been reported in the comments of the post:

1. I'm using version 16.6.0 Preview 3.0 of Visual Studio as recommended in the blog post (`dotnet --version` outputs `3.1.300-preview-015095`).

2. I have cleared my NuGet package cache using `dotnet nuget locals all -c` for good measure.

3. I have installed the latest Blazor WebAssemply template using `dotnet new -i Microsoft.AspNetCore.Components.WebAssembly.Templates::3.2.0-preview4.20210.8`.

4. In Visual Studio I have created a new _Blazor WebAssembly App_ with the following advanced settings:

    - Configure for HTTPS

    - <span>ASP.NET</span> Core hosted

5. After creating the project I use _Start without debugging_. The project builds, IIS Express is started and the app is opened in my browser and works as expected.

6. I then switch the debug profile of the `BlazorApp1.Server` project from `IIS Express` to `BlazorApp1.Server` and then use _Start without debugging_. This time the server starts as a console application and the app is opened in my browser. Blazor loads but `blazor-error-ui` is displayed and the browser console has the following error messages:

    <pre>
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM: Error: Could not find 'getSatelliteAssemblies' in 'window.window.Blazor._internal'.
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at https://localhost:5001/_framework/blazor.webassembly.js:1:9067
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at Array.forEach (<anonymous>)
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at Object.p [as findJSFunction] (https://localhost:5001/_framework/blazor.webassembly.js:1:9028)
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at _mono_wasm_invoke_js_unmarshalled (https://localhost:5001/_framework/wasm/dotnet.3.2.0-preview5.20210.1.js:1:165380)
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at do_icall (<anonymous>:wasm-function[6047]:0x1112f3)
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at do_icall_wrapper (<anonymous>:wasm-function[1900]:0x52abf)
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at interp_exec_method (<anonymous>:wasm-function[1123]:0x2585f)
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at interp_runtime_invoke (<anonymous>:wasm-function[5652]:0xf8d5b)
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at mono_jit_runtime_invoke (<anonymous>:wasm-function[5106]:0xdf5a3)
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:     at do_runtime_invoke (<anonymous>:wasm-function[1412]:0x3d8b6)
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:   at Microsoft.JSInterop.WebAssembly.WebAssemblyJSRuntime.InvokeUnmarshalled[T0,T1,T2,TResult] (System.String identifier, T0 arg0, T1 arg1, T2 arg2) <0x26722c0 + 0x0003a> in <filename unknown>:0 
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:   at Microsoft.AspNetCore.Components.WebAssembly.Services.WebAssemblyJSRuntimeInvoker.InvokeUnmarshalled[T0,T1,T2,TResult] (System.String identifier, T0 arg0, T1 arg1, T2 arg2) <0x25b1ab0 + 0x0001a> in <filename unknown>:0 
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:   at Microsoft.AspNetCore.Components.WebAssembly.Hosting.SatelliteResourcesLoader.LoadSatelliteAssembliesForCurrentCultureAsync () <0x25ebdb8 + 0x00060> in <filename unknown>:0 
    p.printErr @ blazor.webassembly.js:1
    blazor.webassembly.js:1 WASM:   at Microsoft.AspNetCore.Components.WebAssembly.Hosting.EntrypointInvoker.InvokeEntrypoint (System.String assemblyName, System.String[] args, Microsoft.AspNetCore.Components.WebAssembly.Hosting.SatelliteResourcesLoader satelliteResourcesLoader) <0x257d5c0 + 0x000ca> in <filename unknown>:0 
    p.printErr @ blazor.webassembly.js:1
    site.css:1 Failed to load resource: the server responded with a status of 404 ()
    </pre>

It seems that with the latest preview I have to use IIS Express when hosting a Blazor WebAssembly app in an ASP.NET Core website.

**Comment by Liversage** [comment]
I just tested again and I no longer have the issue. I'm pretty sure my computer was restarted after I created the issue so that may explain why the issue has "fixed itself". If only no one else have had the same problem I would immediately close the issue but in the linked blog post other people seem to report problems with the same error messages so I'm not sure if this "Heisenbug" might be worth chasing.
